crates/component-macro/tests/codegen.rs

```diff
@@ -24,3 +24,85 @@ macro_rules! gentest {
 }
 
 component_macro_test_helpers::foreach!(gentest);
+
+mod with_key_and_resources {
+    use anyhow::Result;
+    use wasmtime::component::Resource;
+
+    wasmtime::component::bindgen!({
+        inline: "
+            package demo:pkg
+
+            interface bar {
+                resource a
+                resource b
+            }
+
+            world foo {
+                resource a
+                resource b
+
+                import foo: interface {
+                    resource a
+                    resource b
+                }
+
+                import bar
+            }
+        ",
+        with: {
+            "a": MyA,
+            "b": MyA,
+            "foo/a": MyA,
+            "foo/b": MyA,
+            "demo:pkg/bar/a": MyA,
+            "demo:pkg/bar/b": MyA,
```

Comment:
it doesnt look like the foo/foo/a cases are covered here

Reply:
Oh I don't think that's actually possible via WIT right now since resources are either top-level imports, in an anonymous import, or in an interface which these three cases map to

Reply:
what exercises the `a` from line 46 and `b` from line 47?

Reply:
Ah that's `foo/a` and `foo/b`. The name of the world being "foo" is probably confusing here since the name of the world isn't used for these strings at all. The syntax here is definitely a bit made up by me and isn't exactly specified in WIT at all or anything like that.

Reply:
got it! thank you.